### PR TITLE
Bitcoin traffic component

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do?

Implements Bitcoin mining network traffic simulator
Issue #3411

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] Do all unit tests pass?
* [ ] ~Do all end-to-end tests pass?~
* [ ] ~Any other testing performed?~
    > ~Tested by {Running the Monkey locally with relevant config/running Island/...}~
* [ ] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

**Review by mssalvatore — CHANGES_REQUESTED** [review]
This component should be sending `HTTPRequestEvents`

File: monkey/agent_plugins/payloads/cryptojacker/src/bitcoin_mining_network_traffic_simulator.py [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
 from typing import Optional
 
+import urllib3
+
 from common.event_queue import IAgentEventPublisher
 from common.types import AgentID, SocketAddress
+from common.utils.code_utils import PeriodicCaller
+from infection_monkey.island_api_client.http_client import handle_island_errors
+
+REQUEST_INTERVAL = 10  # seconds
```
Comment (mssalvatore) [inline monkey/agent_plugins/payloads/cryptojacker/src/bitcoin_mining_network_traffic_simulator.py:15]: Have we done any research on what a reasonable value for this is?

File: monkey/agent_plugins/payloads/cryptojacker/src/bitcoin_mining_network_traffic_simulator.py [hunk 2]
```diff
@@ -1,7 +1,21 @@
...
+import random
+import urllib.request
 from typing import Optional
 
+import urllib3
+
 from common.event_queue import IAgentEventPublisher
 from common.types import AgentID, SocketAddress
+from common.utils.code_utils import PeriodicCaller
+from infection_monkey.island_api_client.http_client import handle_island_errors
```
Comment (mssalvatore) [inline monkey/agent_plugins/payloads/cryptojacker/src/bitcoin_mining_network_traffic_simulator.py:13]: It's not a good idea to couple these.

File: monkey/agent_plugins/payloads/cryptojacker/src/bitcoin_mining_network_traffic_simulator.py [hunk 3]
```diff
@@ -15,8 +29,37 @@ def __init__(
...
+            name="Cryptojacking.BitcoinMiningNetworkTrafficSimulator",
+        )
+
     def start(self):
-        pass
+        logger.info("Starting Bitcoin mining network traffic simulator")
+
+        self._send_bitcoin_mining_request_periodically.start()
+
+    @handle_island_errors
```
Comment (mssalvatore) [inline monkey/agent_plugins/payloads/cryptojacker/src/bitcoin_mining_network_traffic_simulator.py:43]: Aside from the fact that we shouldn't couple these components, I don't even think this decorator will work. The http client uses requests, whereas this component is using urllib.

On a separate note, why urllib and not requests? It's fine to leave it as is. Just wondering. For consistency we may want to switch to requests, but up to you.

File: monkey/agent_plugins/payloads/cryptojacker/src/bitcoin_mining_network_traffic_simulator.py (hunk 1 above)
Comment (shreyamalviya) [reply monkey/agent_plugins/payloads/cryptojacker/src/bitcoin_mining_network_traffic_simulator.py:15]: Nope, this is a random value

File: monkey/agent_plugins/payloads/cryptojacker/src/bitcoin_mining_network_traffic_simulator.py (hunk 3 above)
Comment (shreyamalviya) [reply monkey/agent_plugins/payloads/cryptojacker/src/bitcoin_mining_network_traffic_simulator.py:43]: I just picked it up from the script I used for testing when we did the spike, planned on switching to `requests` later but it slipped my mind.

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**